### DISCUSSION TITLE
Understanding solver doesn't converge

### DISCUSSION THREAD

**Discussion by ssingh-ipa** [pybamm-team/PyBaMM#3700] [Q&A]
Hello,

Could you please explain why sometimes the solver finishes a simulation, with the following warning:

At t = 412.623 and h = 8.3058e-18, the corrector convergence failed repeatedly or with |h| = hmin.

How can I understand the impact of this message on my simulation results?  Should I expect the plots to be incorrect at t = 412.623?
Also, could you please refer me to the source code where this warning message is implemented?

**Comment by martinjrobins** [comment]
Hi @ssingh-ipa. This is an internal error message from the Sundials solver (https://github.com/LLNL/sundials) which we use to step forward the DFN/SPM/SPMe equations forward in time. 

The error message that you are getting indicates that the solver has failed to progress with solving the equations at t = 412.623. It has tried to reduce the timestep h, but has reached a lower threshold beyond which it gives up with the error you are seeing.

You should expect the solution returned to stop at t = 412.623, as the solver will not be able to proceed further in time beyond this point
